pkg/scheduler/worker/workerpool.go

```diff
@@ -63,6 +69,10 @@ func (w *Pool) run(ctx context.Context, runOnce bool) error {
 	defer func() {
 		w.logger.Info("Stopping worker pool")
 		workerPool.Release()
+		if err = w.occupRepo.RemoveWorkerPoolOccupancy(w.poolID); err != nil {
+			w.logger.Errorf("Unable to remove worker pool occupancy: %v", err)
+		}
+		w.poolID = "" //do we really need this?
```

Comment:
Can be removed, since when the same Workerpool will be started again, it will get a new uuid, and if not it will be deleted.
```suggestion
```

Reply:
Yup. Totally agree.